### PR TITLE
Add callback to finish exceptions removal before navigation

### PR DESCRIPTION
**Pull request by ionutbedregeanu**
For #5738 
Navigate to the previous screen only after all the exceptions were removed

### CODE REVIEW

File: app/src/main/java/org/mozilla/focus/exceptions/ExceptionsListFragment.kt [hunk 1]
```diff
@@ -121,16 +123,18 @@ open class ExceptionsListFragment : BaseSettingsLikeFragment(), CoroutineScope {
...
 
-            val exceptions = (exceptionList.adapter as DomainListAdapter).selection()
-            TelemetryWrapper.removeAllExceptionDomains(exceptions.count())
-
-            requireComponents.appStore.dispatch(
-                AppAction.NavigateUp(
-                    requireComponents.store.state.selectedTabId
+                val exceptions = (exceptionList.adapter as DomainListAdapter).selection()
+                TelemetryWrapper.removeAllExceptionDomains(exceptions.count())
+                delay(REMOVE_EXCEPTIONS_DELAY)
```
Comment (ionutbedregeanu) [inline app/src/main/java/org/mozilla/focus/exceptions/ExceptionsListFragment.kt:131]: I have mixed feelings about this but I didn't find a better solution without doing changes in AC(maybe that's the only way). I'm open to suggestions if you have 🙏

Comment (Amejia481) [reply app/src/main/java/org/mozilla/focus/exceptions/ExceptionsListFragment.kt:131]: No problem, I'm taking a look to what alternatives we have. I'll report back :)

**Review by Amejia481 — CHANGES_REQUESTED** [review]
I think this approach is bit fragile as we are not completely sure how much time it will take to remove all exceptions in different devices, in some it could take `100 MS` in others it could take more, as a result we will see the same issue on some devices. 

I believe the real problem, it's that the app doesn't have a way to tell when the exceptions have been completely removed. We can expose a callback on AC for the app to tell, similar to [what we are doing for other user cases of tracking protection](https://github.com/mozilla-mobile/android-components/blob/b37f3e46871c201252fdc7107418cf063f1cc20b/components/feature/session/src/main/java/mozilla/components/feature/session/TrackingProtectionUseCases.kt#L145), then we can add loading spinner after the remove all button is clicked and we can show it until the callback tells us the all exceptions have been removed .

**Comment by ionutbedregeanu** [comment]
> I think this approach is bit fragile as we are not completely sure how much time it will take to remove all exceptions in different devices, in some it could take `100 MS` in others it could take more, as a result we will see the same issue on some devices.
> 
> I believe the real problem, it's that the app doesn't have a way to tell when the exceptions have been completely removed. We can expose a callback on AC for the app to tell, similar to [what we are doing for other user cases of tracking protection](https://github.com/mozilla-mobile/android-components/blob/b37f3e46871c201252fdc7107418cf063f1cc20b/components/feature/session/src/main/java/mozilla/components/feature/session/TrackingProtectionUseCases.kt#L145), then we can add loading spinner after the remove all button is clicked and we can show it until the callback tells us the all exceptions have been removed .

I completely agree with you. Thanks, @Amejia481

File: app/src/main/res/layout/fragment_exceptions_domains.xml [hunk 2]
```diff
@@ -38,8 +39,9 @@
         <androidx.recyclerview.widget.RecyclerView
             android:id="@+id/exceptionList"
             android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:orientation="vertical" />
+            android:layout_height="0dp"
+            android:orientation="vertical"
+            android:layout_weight="0.9"/>
```
Comment (Amejia481) [inline app/src/main/res/layout/fragment_exceptions_domains.xml:44]: Is there any specific reason why we are changing the `RecyclerView` layout configurations ? :)

Comment (ionutbedregeanu) [reply app/src/main/res/layout/fragment_exceptions_domains.xml:44]: If we are having many exceptions "remove exceptions" buttons is pushed out of the screen

Comment (Amejia481) [reply app/src/main/res/layout/fragment_exceptions_domains.xml:44]: Thanks for the clarification 👍🏽

**Review by Amejia481 — APPROVED** [review]

**Pull request merged**